File: 2024-maximize-the-confusion-of-an-exam/2024-maximize-the-confusion-of-an-exam.py
class Solution:
    def maxConsecutiveAnswers(self, A: str, k: int) -> int:
        res = 0
        j = 0
        freq = defaultdict(int)
        ma = 0
        for i in range(len(A)):
            freq[A[i]] += 1
            ma = max(ma, freq[A[i]])
            if ma+k < i-j+1:
                freq[A[j]] -= 1
                j += 1
            res = max(res, i-j+1)
        return res

File: 2024-maximize-the-confusion-of-an-exam/2024-maximize-the-confusion-of-an-exam.cpp
class Solution {
public:
    int sol(string& answerKey, int k, char ch) {
        // unordered_map<char, int> m;
        int j = 0;
        int res = 0;
        for (int i = 0; i < answerKey.size(); i++) {
            // m[answerKey[i]] += 1;
            if (answerKey[i] != ch)
                k--;
            while (j < i && k < 0) {
                if (answerKey[j] != ch) {
                    k++;
                }
                j += 1;
            }
            res = max(res, i-j+1);
        }
        return res;
    }
    
    int maxConsecutiveAnswers(string answerKey, int k) {
        return max(sol(answerKey, k, 'T'), sol(answerKey, k, 'F'));
    }
};